_posts/2025-02-13-rfprs.md

```diff
@@ -0,0 +1,372 @@
+Despite this we often will put as little effort as possible into creating PRs. I'd like to convince
+you that even a small investment in improving them pays over-sized dividends to the project and your
+team members.
+
+# Background
+
+I'll start by admitting that because the PR is frequently at the very end of a development effort
+that we should pay respect to the mental exhaustion present. A project spike will start with a
+mountain of known unknowns, and some unknown unknowns as well, all of which an IC will be
+accountable for, in addition to writing code that is maintainable, tested, and covers edge cases. It
+can be exhausting and therefore understandable that we would feel the impulse to create the PR as
+quickly as possible to get this over the finish line. I hope you'll agree that by the end of this
+post that spending a bit more effort to create a better PR is worth it, and is probably not as much
+effort as one might think.
+
+Let's start with the perspective of the reviewer. Everyone is familiar with this scenario: you get
+an notification about a new PR you need to review. You open the PR and see that the title of the PR
+is a JIRA ticket. The description of the PR is blank. You find the JIRA ticket and read it to get
+some context before continuing. Going back to the PR, you see about forty commits, many of which are
+things like "migration" or "wtf please work" or worse. Given this you click the "Files changed" tab
+and see that there are 5k lines of code changed, across 30 files. Resigned to your fate, you then
+try to understand this large changeset all at once. Because it's incredibly difficult to do so and
+because you are also under the gun for your own changesets, you end up reviewing it at the surface
+level and your review mostly consists of style feedback, and perhaps some more nuanced feedback for
+the parts you are able to understand more easily.
+
+> This is clearly not the experience we want.
+
+As the creator of the PR, you are hoping to merge it. But past that, you also want your teammates to
+let you know of things that you may not have considered, or why this particular algorithm you've
+employed may have negative effects on the backend APIs it eventually calls. Style nits are OK but
+avoiding an incident by pointing out an unhandled edge case is much better.
+
+The reviewers only have so much time to devote to your PR. If that time is largely spent being
+mystified about the changeset, that is time that they are not spending giving you the quality
+feedback you should be hoping for. The time a reviewer spends on any particular PR is then
+multiplied by the number of people that end up reviewing it.
+
+> Time spent creating a better PR costs the business less to merge it.
+
+# RFPR Overview
+
+Reviewer First Pull Requests (RFPRs) are a prescriptive approach to building better PRs that are
+cheaper to review. At Mesosphere I worked on a team under [Kevin
```

Comment:
Maybe use a different phrasing than "cheaper"?  Eg: "easier", "that require less cognitive load", ?